index.d.ts

```diff
@@ -0,0 +1,123 @@
+ *    action, and returns a new state. Warped Components makes sure that
+ *    whenever the connected component is mounted, this reducer will act as
+ *    part of the reducers of your store.
+ * @param options.selectors An object whose keys correspond to properties, and values
+ *  are functions that receive the state and return the value for the
+ *  property.
+ * @param options.actions A hash-map of action creators. The underlying component will
+ *  receive them as props, and when called, the resulting action is
+ *  dispatched to the store.
+ * @param options.effects A Cycle application - a function which takes a mapping of
+ *    stream / stream-producers, and returns a mapping of streams.
+ */
+export declare function warped<
+    TSelectors extends Selectors<S>,
+    TActions extends StrMap<ActionCreator>,
+    S = any
+    >(options: {
+        reducer?: redux.Reducer<S, redux.Action<keyof TActions>>
+        selectors: TSelectors,
+        actions: TActions,
+        effects?: CycleMain<WarpedSources<S>, any>
+    }):<C extends react.ComponentType<Partial<WarpedProps<TActions, TSelectors>>>>(component: C) => C;
+
+
+/**
+ * This component does the wiring for your application:
+ *
+ * - Setting up your Redux store, and swapping out its `reducer` as needed.
+ * - Setting up your Cycle application, and swapping out its `main` as needed.
+ * - Linking the Redux Store and Cycle apps by adding two drivers:
+ *     1. `state`: A read-only driver exposing a memory stream of the latest
+ *       Redux state.
+ *     2. `action`: A read-write driver exposing a stream of dispatched
+ *       actions and allowing other actions to be dispatched.
+ *
+ * See https://github.com/wearereasonablepeople/warped-components#WarpedApp
+ */
+export declare const WarpedApp: react.SFC<{
+    initialState?: any,
+    enhancer?: redux.StoreEnhancer,
+    drivers?: StrMap<Driver<Stream<any>, any>>
+}>;
+
+
+/**
+ * Given a mapping of selectors, returns a `mapStateToProps` function, as
+ * accepted by `connect` from React Redux.
+ * See https://github.com/wearereasonablepeople/warped-components#compileSelectors
+ */
+export declare function compileSelectors<S = any>(selectors: Selectors<S>): (state: any, prevProps: any) => any;
+
+/**
+ * Given a mapping of action creators, as returned from
+ * {@link createReducer}, returns a `mapDispatchToProps` function,
+ * as accepted by `connect` from React Redux.
+ * See https://github.com/wearereasonablepeople/warped-components#compileDispatchers
+ */
+export declare function compileDispatchers(actions: StrMap<ActionCreator>): (dispatch: any) => any;
+
+/**
+ * Given an array of reducers, returns a single reducer which transforms the
+ * state by calling all reducers in sequence.
+ * See https://github.com/wearereasonablepeople/warped-components#combineReducers
+ */
+export declare function combineReducers<S = any>(reducers: [redux.Reducer<S>]): redux.Reducer<S>;
```

Comment:
I've never seen this `<S = any>` syntax for generic types. What is it for?

Reply:
It's an optional type param, defaults to `any` in these cases.

Something like https://github.com/reduxjs/redux/blob/master/index.d.ts#L19